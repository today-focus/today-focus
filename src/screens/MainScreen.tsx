import { useState, useEffect } from "react";

import { StatusBar } from "expo-status-bar";
import {
  Keyboard,
  KeyboardAvoidingView,
  Platform,
  StyleSheet,
  Text,
  TouchableHighlight,
  TouchableWithoutFeedback,
  View,
} from "react-native";

import TitleInput from "../components/TitleInput";
import Routine from "../components/Routine";
import TodoList from "../components/TodoList";
import RenderModal from "../components/RenderModal";
import BottomDrawer from "../components/BottomDrawer";

export default function MainScreen() {
  const [date, setDate] = useState<string>("");
  const [modalVisible, setModalVisible] = useState(false);
  const [isModalOkPressed, setIsModalOkPressed] = useState(0);

  useEffect(() => {
    const today = new Date();
    const currentDate = `${today.toLocaleString("en-us", {
      month: "short",
    })}, ${today.getDate()}`;

    setDate(currentDate);
  }, []);

  return (
    <KeyboardAvoidingView
      style={{ flex: 1 }}
      behavior={Platform.OS === "ios" ? "padding" : "height"}
    >
      <View style={styles.container}>
        <StatusBar style="auto" />
        <View style={styles.header}>
          <TouchableHighlight
            onPress={() => console.log("pressed!")}
            underlayColor="#006de9"
            activeOpacity={0.5}
          >
            <Text style={styles.btnText}>{date}</Text>
          </TouchableHighlight>
        </View>
        <View style={styles.contentsContainer}>
          <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
            <View style={styles.todayRoutineContainer}>
              <TitleInput />
              <Routine />
              <TodoList
                isModalOkPressed={isModalOkPressed}
                todoItemList={[
                  {
                    id: Date.now(),
                    text: "",
                    isChecked: false,
                  },
                ]}
              />
              <RenderModal
                modalVisible={modalVisible}
                setModalVisible={setModalVisible}
                setIsModalOkPressed={setIsModalOkPressed}
              />
            </View>
          </TouchableWithoutFeedback>
          <BottomDrawer setModalVisible={setModalVisible} />
        </View>
      </View>
    </KeyboardAvoidingView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#006de9",
  },
  header: {
    flexDirection: "row",
    justifyContent: "center",
    marginTop: 60,
  },
  btnText: {
    fontSize: 28,
    fontWeight: "600",
    color: "#fff",
  },
  contentsContainer: {
    flex: 1.8,
    backgroundColor: "#fff",
    borderTopStartRadius: 40,
    borderTopEndRadius: 40,
    marginTop: 30,
  },
  todayRoutineContainer: {
    flex: 1,
    paddingHorizontal: 20,
  },
});
