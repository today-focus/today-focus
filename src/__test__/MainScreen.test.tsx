import "react-native";
import { render, screen, waitFor } from "@testing-library/react-native";

import renderer from "react-test-renderer";

import MainScreen from "../screens/MainScreen";

describe("Testing MainScreen", () => {
  beforeEach(() => {
    render(<MainScreen />);
  });

  jest.useFakeTimers();
  Date.now = jest.fn(() => 1665373596677);

  it("MainScreen renders correctly", async () => {
    const tree = await waitFor(() => renderer.create(<MainScreen />).toJSON());
    expect(tree).toMatchSnapshot();
  });

  it("MainScreen contains title and subtitle", async () => {
    const title = await screen.findByText(/today's title/i);
    const subtitle = await screen.findByText(/today routine/i);

    expect(title).toBeTruthy();
    expect(subtitle).toBeTruthy();
  });
});
