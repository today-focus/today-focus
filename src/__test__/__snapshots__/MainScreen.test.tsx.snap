// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MainScreen MainScreen renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#006de9",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 60,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 28,
              "fontWeight": "600",
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderTopEndRadius": 40,
          "borderTopStartRadius": 40,
          "flex": 1.8,
          "marginTop": 30,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "paddingHorizontal": 20,
          }
        }
      >
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#006de9",
                "fontSize": 16,
                "fontWeight": "700",
                "marginBottom": 15,
                "marginTop": 35,
                "paddingHorizontal": 20,
              }
            }
          >
            Today's Title
          </Text>
          <TextInput
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="What is your focus for today?"
            returnKeyType="done"
            style={
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#808080",
                "fontSize": 15,
                "justifyContent": "center",
                "marginHorizontal": 20,
                "paddingHorizontal": 5,
                "paddingVertical": 5,
              }
            }
            value=""
          />
        </View>
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#006de9",
                "fontSize": 16,
                "fontWeight": "700",
                "marginBottom": 15,
                "marginTop": 35,
                "paddingHorizontal": 20,
              }
            }
          >
            Today Routine
          </Text>
        </View>
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RCTScrollView
            automaticallyAdjustContentInsets={false}
            data={
              Array [
                Object {
                  "id": 1665373596677,
                  "isChecked": false,
                  "text": "",
                },
              ]
            }
            getItem={[Function]}
            getItemCount={[Function]}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            viewabilityConfigCallbackPairs={Array []}
          >
            <View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "marginHorizontal": 20,
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text />
                  </View>
                  <View
                    style={
                      Object {
                        "paddingVertical": 10,
                      }
                    }
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "borderColor": "#006de9",
                            "borderRadius": 3,
                            "borderWidth": 2,
                            "height": 20,
                            "justifyContent": "center",
                            "width": 20,
                          },
                          false,
                        ]
                      }
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "paddingRight": 5,
                      }
                    }
                  >
                    <TextInput
                      onChangeText={[Function]}
                      onEndEditing={[Function]}
                      onKeyPress={[Function]}
                      onSubmitEditing={[Function]}
                      placeholder="Click Here"
                      returnKeyType="done"
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#fff",
                            "flex": 1,
                            "fontSize": 15,
                            "justifyContent": "center",
                            "marginHorizontal": 5,
                            "paddingHorizontal": 5,
                            "paddingVertical": 10,
                          },
                          false,
                        ]
                      }
                      value=""
                    />
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </RCTSafeAreaView>
        <View />
      </View>
      <View
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Array [
            undefined,
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "bottom": 0,
              "overflow": "hidden",
              "top": 0,
            },
          ]
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "opacity": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0, 109, 233, 0.8)",
              "borderTopLeftRadius": 50,
              "borderTopRightRadius": 50,
              "justifyContent": "center",
              "paddingVertical": 16,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
</View>
`;
